preprocessors/togeojson-gpx.preprocessor.js

```diff
@@ -102,6 +102,14 @@ module.exports = function(infile, outdirectory, callback) {
       });
     });
 
+    var archivedOriginal = path.join(outdirectory, '/archived.gpx');
+    digest(infile, function(err, callback) {
+      fs.writeFile(archivedOriginal, fs.readFile(infile), function(err) {
+        if (err) return callback(err);
+        return archivedOriginal;
```

Comment:
`fs.writeFile()` doesn't expect anything specific returned. I assume you were modeling your code [based on how the metadata file is written](https://github.com/mapbox/preprocessorcerer/blob/45d9f4a91153625af3748cb0c6f9a13f480453bc/preprocessors/togeojson-gpx.preprocessor.js#L96-L103)):

> return createIndices(callback);

One important factor to notice here is that a function is being returned, called `createdIndices()`, and look what it's being passed as a parameter...`callback`. This is the same callback function [passed into the main function call](https://github.com/mapbox/preprocessorcerer/blob/45d9f4a91153625af3748cb0c6f9a13f480453bc/preprocessors/togeojson-gpx.preprocessor.js#L18). The reason for this is because `createIndices()` was previously planned to be the _last step_ within this processor. And since creating indexes are implemented in a standalone function (in order to separate specific behaviour), `createIndices()` can simply [return the original callback](https://github.com/mapbox/preprocessorcerer/blob/45d9f4a91153625af3748cb0c6f9a13f480453bc/preprocessors/togeojson-gpx.preprocessor.js#L122) when it's finished or errors.

So, with your new changes, creating indexes will no longer be the last step in the processor. Instead, archiving the original file will be the last and final step, and therefore will need to return the initial callback passed into the main function once archiving is finished.